src/Microsoft.OpenApi.OData.Reader/Operation/ODataTypeCastGetOperationHandler.cs

```diff
@@ -187,79 +195,17 @@ protected override void SetBasicInfo(OpenApiOperation operation)
 
 	/// <inheritdoc/>
 	protected override void SetResponses(OpenApiOperation operation)
-	{
-		if (IsSingleElement)
-			SetSingleResponse(operation);
+    {
+        if (IsSingleElement)
+			SetSingleResponse(operation, TargetSchemaElement.FullName());
 		else
-			SetCollectionResponse(operation);
+			SetCollectionResponse(operation, TargetSchemaElement.FullName());
 
 		operation.AddErrorResponses(Context.Settings, false);
 
 		base.SetResponses(operation);
 	}
 
-	private void SetCollectionResponse(OpenApiOperation operation)
-	{
-		operation.Responses = new OpenApiResponses
-		{
-			{
-				Context.Settings.UseSuccessStatusCodeRange ? Constants.StatusCodeClass2XX : Constants.StatusCode200,
-				new OpenApiResponse
-				{
-					UnresolvedReference = true,
-					Reference = new OpenApiReference()
-					{
-						Type = ReferenceType.Response,
-						Id = $"{TargetSchemaElement.FullName()}{Constants.CollectionSchemaSuffix}"
-					},
-				}
-			}
-		};
-	}
-
-	private void SetSingleResponse(OpenApiOperation operation)
-	{
-		OpenApiSchema schema = null;
-
-		if (Context.Settings.EnableDerivedTypesReferencesForResponses)
-		{
-			schema = EdmModelHelper.GetDerivedTypesReferenceSchema(targetStructuredType, Context.Model);
-		}
```

Comment:
Out of curiosity, is this functionality lost with the latest change? or is this not needed?

Reply:
Thanks for pointing this out. I had actually missed this